docs/reference/db-config.rst

```diff
@@ -1,8 +1,28 @@
 :orphan:
 
-=======================
-db/config.cc properties
-=======================
+======================
+db/config.cc reference
+======================
 
-.. datatemplate:yaml:: ../_data/db_config.yaml
+Configuration properties
+------------------------
+
+.. dbconfigdatatemplate:: ../_data/db_config.yaml
   :template: db_config.tmpl
+  :filter: Used
+
+
+Deprecated properties
+---------------------
+
+As ScyllaDB evolves, certain configuration properties may become obsolete or redundant.
+In our commitment to ensure smooth transitions, we have opted to keep this properties documented.
+Even though ScyllaDB does not use their values, we are maintaining them for anyone using older configurations that reference these options.
```

Comment:
If I understand correctly, you are talking here about "Unused" properties. I wouldn't call those "deprecated". We have a couple of those too, and it's different. a "deprecated" feature means it's still fully supported in this version but we plan to remove it in a future version. an "Unused" option is already not supported in this version.

If this is the case and you're talking about "Unused", I think the word "deprecated" is the wrong word, and also here, "maintaining them" isn't clear - I think we should be clearer that we continue to allow these configuration options to be specified (without complaining that they are unknown options) - but outright ignore their value. We do **not** maintain the old code that did anything with the value of these "Unused" properties.

Reply:
I see in the following patches that you already understood this and fixed most of the problems. Please conflate all the patches into one so I can review the final version and not an incorrect version and its fixes.